lib/ProductOpener/Ingredients.pm

```diff
@@ -2101,7 +2101,7 @@ sub parse_ingredients_text_service ($product_ref, $updated_product_fields_ref) {
 		my @ingredients = ();
 
 		# 2 known ingredients separated by "and" ?
-		if ($before =~ /$and/i) {
+		if (($before =~ /$and/i) and (not defined $percent_or_quantity_value)) {
```

Comment:
If we do that, we fix the error with the percent being applied twice, but we introduce an issue: the ingredients won't be recognized.

An option could be to add a "compound" ingredient, as discussed in the comments of https://github.com/openfoodfacts/openfoodfacts-server/pull/9236

in practice it would then result in "[compound ingredients] 80% (pulpe de tomates, puree de tomates)"